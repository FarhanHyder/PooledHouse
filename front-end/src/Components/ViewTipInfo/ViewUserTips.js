/**
 * @file ViewUserTips component to display all the tips data by the userInfo
 * @class ViewUserTips 
 */

import React from 'react';
import { Table, Container } from 'react-bootstrap';

const ViewUserTips = (props) => {
    let userTips =  props.tipInfo.filter(tips => tips.user === props.user);
    
    let tipInfo = (
        <Container fluid>
            <div>
            <Table>
                <thead>
                    <tr>
                        <th>Business</th>
                        <th>Neighborhood</th>
                        <th>Shift</th>
                        <th>Takehome</th>
                        <th>Shift Length</th>
                        <th>Shift Time</th>
                        <th>Shift Date</th>
                    </tr>
                </thead>
                <tbody>
                    { Object.keys(userTips).map(key => {
                        return (
                            <tr>
                                <td>{ userTips[key].business_name }</td>
                                <td>{ userTips[key].neighborhood }</td>
                                <td>{ userTips[key].shift_position }</td>
                                <td>{ userTips[key].takehome }</td>
                                <td>{ userTips[key].shift_length }</td>
                                <td>{ userTips[key].shift_time }</td>
                                <td>{ userTips[key].shift_date }</td>
                                
                            </tr>
                        )
                    }) }
                </tbody>
            </Table>
            </div>
        </Container>
    );

    return tipInfo;
}

export default ViewUserTips;