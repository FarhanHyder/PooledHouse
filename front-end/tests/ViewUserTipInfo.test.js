import React from 'react';
import renderer from 'react-test-renderer';
import ViewUserTips from '../src/Components/ViewTipInfo/ViewUserTipInfo';

test('ViewUserTips should render correct for single element', () => {
  const thisProps = [{
    'user': 'shofi',
    'shift_length': 5,
    'shift_time': 'PM',
    'shift_date': '04/30/2019',
    'shift_position': 'Busser',
    'takehome': 87,
    'business_name': 'Poolinos',
    'business_street_address': '292 Bowery',
    'business_city': 'New York',
    'business_state': 'NY',
    'business_zip': '10012',
    'neighborhood': 'Soho'
  }];

  const component = renderer.create(
    <ViewUserTips tipInfo={thisProps} user='shofi' />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('ViewUserTips should render correct for multiple elements', () => {
  const thisProps = [{
    'user': 'shofi',
    'shift_length': 5,
    'shift_time': 'PM',
    'shift_date': '04/30/2019',
    'shift_position': 'Busser',
    'takehome': 87,
    'business_name': 'Poolinos',
    'business_street_address': '292 Bowery',
    'business_city': 'New York',
    'business_state': 'NY',
    'business_zip': '10012',
    'neighborhood': 'Soho'
  },
  {
    'user': 'shofi',
    'shift_length': 6,
    'shift_time': 'PM',
    'shift_date': '05/02/2019',
    'shift_position': 'Busser',
    'takehome': 93,
    'business_name': 'Poolinos',
    'business_street_address': '292 Bowery',
    'business_city': 'New York',
    'business_state': 'NY',
    'business_zip': '10012',
    'neighborhood': 'Soho'
  },
  {
    'user': 'carl',
    'shift_length': 4,
    'shift_time': 'PM',
    'shift_date': '04/30/2019',
    'shift_position': 'Busser',
    'takehome': 75,
    'business_name': 'Balthazer',
    'business_street_address': '292 Bowery',
    'business_city': 'New York',
    'business_state': 'NY',
    'business_zip': '10012',
    'neighborhood': 'Soho'
  }];

  const component = renderer.create(
    <ViewUserTips tipInfo={thisProps} user='shofi' />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
