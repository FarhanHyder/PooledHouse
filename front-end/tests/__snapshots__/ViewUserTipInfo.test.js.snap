// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewUserTips should render correct for multiple elements 1`] = `
Array [
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Business
          </th>
          <th>
            Neighborhood
          </th>
          <th>
            Shift
          </th>
          <th>
            Takehome
          </th>
          <th>
            Shift Length
          </th>
          <th>
            Shift Time
          </th>
          <th>
            Shift Date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            Poolinos
          </th>
          <th>
            Soho
          </th>
          <th>
            Busser
          </th>
          <th>
            87
          </th>
          <th>
            5
          </th>
          <th>
            PM
          </th>
          <th>
            04/30/2019
          </th>
        </tr>
      </tbody>
    </table>
  </div>,
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Business
          </th>
          <th>
            Neighborhood
          </th>
          <th>
            Shift
          </th>
          <th>
            Takehome
          </th>
          <th>
            Shift Length
          </th>
          <th>
            Shift Time
          </th>
          <th>
            Shift Date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th>
            Poolinos
          </th>
          <th>
            Soho
          </th>
          <th>
            Busser
          </th>
          <th>
            93
          </th>
          <th>
            6
          </th>
          <th>
            PM
          </th>
          <th>
            05/02/2019
          </th>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;

exports[`ViewUserTips should render correct for single element 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Business
        </th>
        <th>
          Neighborhood
        </th>
        <th>
          Shift
        </th>
        <th>
          Takehome
        </th>
        <th>
          Shift Length
        </th>
        <th>
          Shift Time
        </th>
        <th>
          Shift Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          Poolinos
        </th>
        <th>
          Soho
        </th>
        <th>
          Busser
        </th>
        <th>
          87
        </th>
        <th>
          5
        </th>
        <th>
          PM
        </th>
        <th>
          04/30/2019
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
